src/Analyzers/CSharp/Analyzers/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs

```diff
@@ -169,9 +168,15 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             var nextToken = expressionStatement.GetLastToken().GetNextToken();
 
             // Fade out the code up to the expression statement.
-            syntaxContext.ReportDiagnostic(Diagnostic.Create(UnnecessaryWithSuggestionDescriptor,
-                Location.Create(tree, TextSpan.FromBounds(firstStatement.SpanStart, previousToken.Span.End)),
-                additionalLocations, properties));
+            var fadeLocation = Location.Create(tree, TextSpan.FromBounds(firstStatement.SpanStart, previousToken.Span.End));
+            syntaxContext.ReportDiagnostic(DiagnosticHelper.CreateWithLocationTags(
+                Descriptor,
+                fadeLocation,
+                ReportDiagnostic.Default,
+                additionalLocations.Add(fadeLocation),
+                tagIndices: ImmutableDictionary<string, IEnumerable<int>>.Empty
+                    .Add(nameof(WellKnownDiagnosticTags.Unnecessary), SpecializedCollections.SingletonEnumerable(additionalLocations.Length)),
```

Comment:
not certain i get what this means.  why is the 'additionalLocations.Length' being added in here.

Reply:
oh... i think i see.  we do `additionalLocations.Add(fadeLocation)` which produces a fresh array, with the fadeLocation at the end.  and we then pass in `additionalLocations.Length` to say where that index is.  but .Length refers to the original locations-array, prior to the add.  

gotta say, i don't love this :)   but it's isolates enough i suppose it's not a big deal.

Reply:
➡️ It'll still be one collection, but this will be simplified as part of https://github.com/dotnet/roslyn/pull/46666#discussion_r470892118